packages/graphql/src/utils/execute.ts

```diff
@@ -88,40 +136,25 @@ async function execute(input: {
     try {
         debug("%s", `About to execute Cypher:\nCypher:\n${cypher}\nParams:\n${JSON.stringify(input.params, null, 2)}`);
 
-        const app = `${environment.NPM_PACKAGE_NAME}@${environment.NPM_PACKAGE_VERSION}`;
-
-        let result: QueryResult | undefined;
-        const transactionWork = (tx: Transaction) => tx.run(cypher, input.params);
-        const transactionConfig = {
-            metadata: {
-                app,
-                type: "user-transpiled",
-            },
-        };
-
-        switch (input.defaultAccessMode) {
-            case "READ":
-                result = await session.readTransaction(transactionWork, transactionConfig);
-                break;
-            case "WRITE":
-                result = await session.writeTransaction(transactionWork, transactionConfig);
-                break;
-            // no default
-        }
+        const result: QueryResult | undefined = await executor.transaction.run(cypher, input.params);
 
         if (!result) {
             throw new Error("Unable to execute query against Neo4j database");
         }
 
+        if (executor.session) {
```

Comment:
Why checking for `executor.session` here if the conditional commits the transaction?

Reply:
Same reason as above basically. If a session exists, it means that the transaction was opened by the execute function, and thus should also be closed by it.

Reply:
Makes sense! Perhaps a similar variable to what I suggested above?

Reply:
Yes, that would work :+1: